tests/test_threadtree.py

```diff
@@ -0,0 +1,40 @@
+from easypy.concurrency import concurrent
+from easypy.threadtree import walk_frames
+
+
+def collect_frames():
+    """Collect only interesting frames"""
+    result = []
+
+    for frame in walk_frames(across_threads=True):
+        if (
+            "python" in str(frame)
+            or "concurrency" in str(frame)
+            or "/threadtree.py" in str(frame)
+        ):
+            continue
+        result.append(frame)
+
+    return result
+
+
+def test_walk_frames():
+    frames = []
+    def func_b():
+        nonlocal frames
+        frames = collect_frames()
+    def func_a():
+        with concurrent(func_b):
+            pass
+
+    with concurrent(func_a):
+        pass
```

Comment:
We need the test to allow, in-fact - make sure, that the outer thread as moved on when the inner thread is collecting the frames. I would use an event object to tell the inner thread when it can collect the fames.